### PR TITLE
[PFX-583] - @gasket/plugin-otel

### PR DESCRIPTION
**Pull request by mmason2-godaddy**
## Summary
Initial release of the OTel Gasket plugin.  Testing and development was done using the Gasket v7 changes and the `@gasket/preset-api` preset.

Metrics are currently not supported by Elastic when using a OTel Metrics Exporter - [Limitation](https://www.elastic.co/guide/en/observability/current/apm-open-telemetry-known-limitations.html#apm-open-telemetry-metrics-limitations). The functionality is implemented in the instrumentation file but commented out. I've confirmed that the metrics data is being sent successfully via the gRPC exporter but it's not able to be processed and viewed in Kibana currently.

Log transports are working correctly and are sent to ESS when using the Gasket logger interface. Testing was done with the new v7 logging plugins - `@gasket/plugin-logger` & `@gasket/plugin-winston`.

Traces are being exported properly and are available in ESS.

### Local Testing
```
npm i -g create-gasket-app@7.0.0-next.21 

create-gasket-app my-app --presets @gasket/preset-api@7.0.0-next.21 

cd ./my-app

npm i @gasket/plugin-otel
// Follow the rest of the README
```

### Follow Up

- Test the Gasket v7 changes with NextJs 14 and OTel to ensure that traces are correct in ESS(currently blocked but the resolution is close to completion)

## Changelog
Initial release

## Test Plan
Tests added

### CODE REVIEW

**Comment by mmason2-godaddy** [comment]
I have an ESS instance to show the logs and traces if that's necessary for the approval 👍. I can also demo to the team.

File: packages/gasket-plugin-otel/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+```shell
+OTEL_EXPORTER_OTLP_ENDPOINT=<your-ess-otel-endpoint>
+OTEL_EXPORTER_OTLP_HEADERS="Authorization=Bearer <ess-secret>"
+OTEL_SERVICE_NAME=<service-name>
+
+# Optional
+OTEL_SERVICE_VERSION=<service-version>
+OTEL_NODE_ENABLED_INSTRUMENTATIONS=<instrumentations>
+OTEL_NODE_RESOURCE_DETECTORS=<resource-detectors>
```
Comment (hmohamed-godaddy) [inline packages/gasket-plugin-otel/README.md:25]: this is really exciting. Where would be find out ess otel endpoint? Referencing those docs here would be helped.

Comment (mmason2-godaddy) [reply packages/gasket-plugin-otel/README.md:25]: Another thing to note, these values are used during deploys from ASM:

```diff
- --build-arg ELASTIC_APM_SERVER_URL=${{ steps.get-secrets.outputs.essp_apm_deployment_apm_server_url }} \
- --build-arg ELASTIC_APM_SECRET_TOKEN=${{ steps.get-secrets.outputs.essp_apm_deployment_apm_secret_token }} \

+ --build-arg OTEL_EXPORTER_OTLP_ENDPOINT=${{ steps.get-secrets.outputs.essp_apm_deployment_apm_server_url }} \
+ --build-arg OTEL_EXPORTER_OTLP_HEADERS="Authorization=Bearer ${{ steps.get-secrets.outputs.essp_apm_deployment_apm_secret_token }}" \
```

Comment (hmohamed-godaddy) [reply packages/gasket-plugin-otel/README.md:25]: would the endpoint end in `https://<your-ess-instance>/api/apm/otel`? when you note that it is `your-ess-otel-endpoint` I thought there was a specific route for the endpoint.

**Pull request closed without merging**